### ISSUE TITLE
Unable to build react native reanimated

### ISSUE THREAD

**Issue by gift-sequoiaat** [software-mansion/react-native-reanimated#4668]
### Description

While Installing and building react native reanimated version 3.0 and 3.1 on react native version 0.72 the build fails with following error

A problem was found with the configuration of task ':react-native-reanimated:mergeDebugJniLibFolders' (type 'MergeSourceSetFolders').
  - Gradle detected a problem with the following location: '/home/giftson/react-native-test/AwesomeProject/node_modules/react-native-reanimated/android/src/main/jniLibs'.
    
    Reason: Task ':react-native-reanimated:mergeDebugJniLibFolders' uses this output of task ':react-native-reanimated:packageNdkLibs' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
    
    Possible solutions:
      1. Declare task ':react-native-reanimated:packageNdkLibs' as an input of ':react-native-reanimated:mergeDebugJniLibFolders'.
      2. Declare an explicit dependency on ':react-native-reanimated:packageNdkLibs' from ':react-native-reanimated:mergeDebugJniLibFolders' using Task#dependsOn.
      3. Declare an explicit dependency on ':react-native-reanimated:packageNdkLibs' from ':react-native-reanimated:mergeDebugJniLibFolders' using Task#mustRunAfter.
    
    Please refer to https://docs.gradle.org/8.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 46s

### Steps to reproduce

- Install reanimated 

### Snack or a link to a repository

https://github.com/gift-sequoiaat/AwesomeProject.git

### Reanimated version

3.0.0

### React Native version

0.72

### Platforms

Android

### JavaScript runtime

None

### Workflow

None

### Architecture

None

### Build type

None

### Device

None

### Acknowledgements

Yes

**Comment by gift-sequoiaat** [comment]
> Hey! 👋
> 
> The issue doesn't seem to contain a [minimal reproduction](https://stackoverflow.com/help/minimal-reproducible-example).
> 
> Could you provide a [snack](https://snack.expo.dev/) or a link to a GitHub repository under your username that reproduces the problem?

https://github.com/gift-sequoiaat/AwesomeProject.git

**Comment by tomekzaw** [comment]
Hey @gift-sequoiaat, thanks for submitting this issue.

Reanimated 3.1.0 was released like 2 months ago and unfortunately it does no longer work with React Native 0.72 due to a number of breaking changes introduced in the framework itself.

However, worry not, as recently released Reanimated 3.3.0 which fully supports React Native 0.72.

**Comment by gift-sequoiaat** [comment]
Hi @tomekzaw thanks for the quick reply but i need reanimated library for react native vision camera can i please know is there any plans to support reanimated v2 for RN 0.72

**Comment by VictorFDiniz** [comment]
I am having problems with reanimated. I started my project last month and for over a week I have been trying to implement the drawer but reanimated always gives some error. I have tested several combinations of versions between reanimated and react but it always generates some error in the build. When I remove reanimated from the code everything works normally. These are the versions of my dependencies:

  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "lint": "eslint .",
    "start": "react-native start",
    "test": "jest"
  },
  "dependencies": {
    "@react-navigation/drawer": "^6.6.3",
    "@react-navigation/native": "^6.1.7",
    "@react-navigation/native-stack": "^6.9.13",
    "react": "18.2.0",
    "react-native": "0.72.3",
    "react-native-gesture-handler": "^2.12.0",
    "react-native-reanimated": "3.3.0",
    "react-native-responsive-fontsize": "^0.5.1",
    "react-native-safe-area-context": "^4.7.1",
    "react-native-screens": "^3.22.1",
    "react-native-swiper": "^1.6.0",
    "react-native-vector-icons": "^9.2.0"
  },
  "devDependencies": {
    "@babel/core": "^7.20.0",
    "@babel/preset-env": "^7.20.0",
    "@babel/runtime": "^7.20.0",
    "@react-native/eslint-config": "^0.72.2",
    "@react-native/metro-config": "^0.72.9",
    "@tsconfig/react-native": "^3.0.0",
    "@types/react": "^18.0.24",
    "@types/react-native-vector-icons": "^6.4.13",
    "@types/react-test-renderer": "^18.0.0",
    "babel-jest": "^29.2.1",
    "eslint": "^8.19.0",
    "jest": "^29.2.1",
    "metro-react-native-babel-preset": "0.76.7",
    "prettier": "^2.4.1",
    "react-test-renderer": "18.2.0",
    "typescript": "4.8.4"
  },
  "engines": {
    "node": ">=16"
  }
}

The error:

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':react-native-reanimated:configureCMakeDebug[arm64-v8a]'.
> [CXX1428] exception while building Json A problem occurred starting process 'command 'C:\Users\victo\AppData\Local\Android\Sdk\cmake\3.22.1\bin\cmake.exe''

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3m 20s
error Failed to install the app. Command failed with exit code 1: gradlew.bat app:installDebug -PreactNativeDevServerPort=8081
Note: Some input files use or override a deprecated API. Note: Recompile with -Xlint:deprecation for details. Note: Some input files use unchecked or unsafe operations. Note: Recompile with -Xlint:unchecked for details. FAILURE: Build failed with an exception. * What went wrong: Execution failed for task ':react-native-reanimated:configureCMakeDebug[arm64-v8a]'. > [CXX1428] exception while building Json A problem occurred starting process 'command 'C:\Users\victo\AppData\Local\Android\Sdk\cmake\3.22.1\bin\cmake.exe'' * Try: > Run with --stacktrace option to get the stack trace. > Run with --info or --debug option to get more log output. > Run with --scan to get full insights. * Get more help at https://help.gradle.org BUILD FAILED in 3m 20s.
info Run CLI with --verbose flag for more details.

**Comment by Arjan-Zuidema** [comment]
@VictorFDiniz I also had this problem. I fixed it by uninstalling cmake 3.22.1 and installing cmake 3.18.1 in Android studio

**Comment by VictorFDiniz** [comment]
> @VictorFDiniz I also had this problem. I fixed it by uninstalling cmake 3.22.1 and installing cmake 3.18.1 in Android studio

Worked, thank you :)

**Comment by krystofcelba** [comment]
> Hi @tomekzaw thanks for the quick reply but i need reanimated library for react native vision camera can i please know is there any plans to support reanimated v2 for RN 0.72

For anyone encountering a build issue with a React Native 0.72 project using react-native-reanimated 2.17.0, I managed to successfully build the project by adding the following snippet to node_modules/react-native-reanimated/android/build.gradle:
```
gradle
Copy code
tasks.withType(com.android.build.gradle.tasks.MergeSourceSetFolders) {
    compileTask ->
        compileTask.dependsOn(packageNdkLibs)
}
```
Below is a patch for the patch-package, which includes the necessary modifications for a successful iOS build.

https://gist.github.com/krystofcelba/4afea77d446df5a604e697fd96a359fa
